execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -394,6 +479,238 @@ where
+                        new_elements.push((trie_key, trie_node));
+                        break;
+                    }
+                    // The sibling is a leaf and the grandparent is a node.  Reseat the single leaf
+                    // sibling into the grandparent.
+                    (Pointer::LeafPointer(..), Some((idx, Trie::Node { mut pointer_block }))) => {
+                        pointer_block[idx as usize] = Some(sibling_pointer);
+                        let trie_node = Trie::Node { pointer_block };
+                        let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                        new_elements.push((trie_key, trie_node))
+                    }
+                    // The sibling is a leaf and the grandparent is an extension.
+                    (Pointer::LeafPointer(..), Some((_, Trie::Extension { .. }))) => {
+                        match parents.pop() {
+                            None => panic!("Root node cannot be an extension node"),
+                            Some((_, Trie::Leaf { .. })) => panic!("Should not find leaf"),
+                            Some((_, Trie::Extension { .. })) => {
+                                panic!("Extension cannot extend to an extension")
+                            }
+                            // The great-grandparent is a node. Reseat the single leaf sibling into
+                            // the position the grandparent was in.
+                            Some((idx, Trie::Node { mut pointer_block })) => {
+                                pointer_block[idx as usize] = Some(sibling_pointer);
+                                let trie_node = Trie::Node { pointer_block };
+                                let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                                new_elements.push((trie_key, trie_node))
+                            }
+                        }
+                    }
+                    // The single sibling is a node or an extension, and a grandparent exists.
+                    // Therefore the parent is not the root
+                    (Pointer::NodePointer(sibling_trie_key), Some((idx, grandparent))) => {
+                        // Push the grandparent back onto the parents so it may be processed later.
+                        parents.push((idx, grandparent));
+                        // Elsewhere we maintain the invariant that all trie keys have corresponding
+                        // trie values.
+                        let sibling_trie = store
+                            .get(txn, &sibling_trie_key)?
+                            .expect("should have sibling");
+                        match sibling_trie {
+                            Trie::Leaf { .. } => {
+                                panic!("Node pointer should not point to leaf")
+                            }
+                            // The single sibling is a node, and the grandparent does not exist.
```

Comment:
It doesn't? Didn't we just push it back onto `parents`?

Reply:
Yes, this comment predated that push, we will fix it.

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/1155/commits/e0718bb1afd75d046aa458c8647f39714531e3b1